app/services/foreman/renderer/scope/macros/helpers.rb

```diff
@@ -155,6 +155,32 @@ def truthy?(value = nil)
           def falsy?(value = nil)
             Foreman::Cast.to_bool(value) == false
           end
+
+          apipie :method, 'Generate a web request' do
+            required :utility, String, desc: 'The utility to use for the web request ("curl" or "wget").'
+            required :url, String, desc: 'The URL for the web request.'
+            optional :ssl_ca_cert, String, desc: 'The path to the SSL certificate.'
+            optional :headers, Array, of: String, desc: 'Array containing the headers for the web request starting with "--header".'
+            optional :params, Array, of: String, desc: 'Array containing the POST parameters for the web request.'
+            optional :output_file, String, desc: 'The path were the result of the web request will be stored.'
```

Comment:
There will still be a minor issue (bug in ApipieDSL), but it will still be more correct:
```suggestion
            keyword :utility, String, desc: 'The utility to use for the web request ("curl" or "wget").'
            keyword :url, String, desc: 'The URL for the web request.'
            keyword :ssl_ca_cert, String, desc: 'The path to the SSL certificate.', default: nil
            keyword :headers, Array, of: String, desc: 'Array containing the headers for the web request starting with "--header".', default: nil
            keyword :params, Array, of: String, desc: 'Array containing the POST parameters for the web request.', default: nil
            keyword :output_file, String, desc: 'The path were the result of the web request will be stored.', default: nil
```

Reply:
What minor issue do you mean?
I found out that with this change the documentation of the `generate_web_request` method now states that all parameters are optional and have the default value `nil` whereas with `required` and `optional` the parameters appeared correctly in the documentation. Is this the bug you mean?

Reply:
Yeap. The suggested changes are still valid, it's just I need to update ApipieDSL with a fix for that misleading :/